execution-plan/src/arithmetic.rs

```diff
@@ -55,6 +55,30 @@ impl UnaryArithmetic {
     }
 }
 
+trait Power {
+    fn power(self, other: Self) -> Self;
+}
+
+macro_rules! power_float_impl {
+    ($($t:ty)*) => ($(
+        impl Power for $t {
+            fn power(self, other: $t) -> $t { self.powf(other) }
+        }
+
+    )*)
+}
+power_float_impl! { f32 f64 }
+
+macro_rules! power_int_impl {
+    ($($t:ty)*) => ($(
+        impl Power for $t {
+            fn power(self, other: $t) -> $t { self.overflowing_pow(other as u32).0 }
+        }
+
+    )*)
+}
+power_int_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
```

Comment:
Could you please add some unit tests to check the behaviour with negative numbers?